confgenerator/confgenerator_test.go

```diff
@@ -48,94 +46,119 @@ var (
-		hostInfo.PlatformVersion = "linux_platform_version"
-	}
-}
-
-func TestGenerateConfsWithValidInput(t *testing.T) {
-	dirPath := validTestdataDir + "/" + platform
-	logsDir := defaultLogsDir
-	stateDir := defaultStateDir
-	if platform == "windows" {
-		logsDir = windowsDefaultLogsDir
-		stateDir = windowsDefaultStateDir
+var hostInfo = func() *host.InfoStat {
+	//In order to make test data static, we put static value for platform-wise fields.
+	if runtime.GOOS == "windows" {
+		return &host.InfoStat{
+			OS:              "windows",
+			Platform:        "win_platform",
+			PlatformVersion: "win_platform_version",
+		}
 	}
-	dirs, err := ioutil.ReadDir(dirPath)
-	if err != nil {
-		t.Fatal(err)
+	return &host.InfoStat{
+		OS:              "linux",
+		Platform:        "linux_platform",
+		PlatformVersion: "linux_platform_version",
 	}
+}()
 
-	for _, d := range dirs {
-		testName := d.Name()
-		t.Run(testName, func(t *testing.T) {
-			unifiedConfigFilePath := fmt.Sprintf(dirPath+"/%s/input.yaml", testName)
-			// Special-case the default config.  It lives directly in the
-			// confgenerator directory.  The golden files are still in the
-			// testdata directory.
-			if d.Name() == "default_config" {
-				unifiedConfigFilePath = "default-config.yaml"
-			} else if d.Name() == "windows_default_config" {
-				unifiedConfigFilePath = "windows-default-config.yaml"
-			}
+type platformConfig struct {
+	defaultLogsDir  string
+	defaultStateDir string
+	defaultConfig   string
+	*host.InfoStat
+}
 
-			data, err := ioutil.ReadFile(unifiedConfigFilePath)
-			if err != nil {
-				t.Fatalf("ReadFile(%q) got %v", unifiedConfigFilePath, err)
-			}
-			uc, err := ParseUnifiedConfig(data)
-			if err != nil {
-				t.Fatalf("ParseUnifiedConfig got %v", err)
-			}
+var platforms = []platformConfig{
+	platformConfig{
+		defaultLogsDir:  "/var/log/google-cloud-ops-agent/subagents",
+		defaultStateDir: "/var/lib/google-cloud-ops-agent/fluent-bit",
+		defaultConfig:   "default-config.yaml",
+		InfoStat: &host.InfoStat{
+			OS:              "linux",
+			Platform:        "linux_platform",
+			PlatformVersion: "linux_platform_version",
+		},
+	},
+	platformConfig{
+		defaultLogsDir:  `C:\ProgramData\Google\Cloud Operations\Ops Agent\log`,
+		defaultStateDir: `C:\ProgramData\Google\Cloud Operations\Ops Agent\run`,
+		defaultConfig:   "windows-default-config.yaml",
+		InfoStat: &host.InfoStat{
+			OS:              "windows",
+			Platform:        "win_platform",
+			PlatformVersion: "win_platform_version",
+		},
+	},
+}
 
-			// Retrieve the expected golden conf files.
-			expectedMainConfig := readFileContent(testName, goldenMainPath, t, true)
-			expectedParserConfig := readFileContent(testName, goldenParserPath, t, true)
-			// Generate the actual conf files.
-			mainConf, parserConf, err := uc.GenerateFluentBitConfigs(logsDir, stateDir, hostInfo)
+func TestGenerateConfsWithValidInput(t *testing.T) {
+	for _, platform := range platforms {
```

Comment:
Another thing to be careful of if we ever decide to run these tests in parallel: https://golang.org/doc/faq#closures_and_goroutines.

Reply:
I've hardened this by removing any pointers from the `platformConfig` struct. I believe this means that each test will run with a copy of the values we care about. Additionally, the inner tests were dependent on a `testName` variable which I believe was exhibiting the behavior, so I replaced that.

I understand the point you're making, and acknowledge that this could be reworked further to protect against future changes. However, I suspect this is sufficient for now. Let me know if you disagree.

In any case, I've marked all `confgenerator` tests as parallel, and there doesn't appear to be an immediate problem, at least in my local environment:
```
> go test -count=10000 -timeout=5m -mod=mod github.com/GoogleCloudPlatform/ops-agent/confgenerator
ok      github.com/GoogleCloudPlatform/ops-agent/confgenerator  78.127s
```

Reply:
Well, I see that code coverage has dropped substantially, so I think I understood the implications less well than I expected. Looking into it...

Reply:
Confirmed that this was doing exactly what you warned me about. 

However, I actually get it now. Should be good to go.

Reply:
Heh, I wasn't necessarily suggesting to make the tests parallel, just avoid painting ourselves into the corner so we couldn't make them parallel in the future. But this is fine too.